### ISSUE TITLE
Can not create new blocks

### ISSUE THREAD

**Issue by privaterra** [civicrm/org.civicrm.contactlayout#94]
I've just installed Contact layout Editor and can't get the Add Block option to work. When I click on the open, a new - blank - window opens up. The error seems to be similar to what has been [reported](https://civicrm.stackexchange.com/questions/38704/contact-layout-editor-create-new-block-not-working) back in Feb.

Would appreciate any suggestions on what I might be able to do to get the extension to work properly.

Here's my install details...

Versions:
- CiviCRM 5.36.0.
- Drupal 9.1.6
- Angular Profiles : 1.22
- Contact Layout Editor:  1.8.2

**Comment by colemanw** [comment]
Can you open up your browser console, then click the "Add Block" button & copy-paste any errors here?

**Comment by privaterra** [comment]
Thanks for the quick reply!

Below is the content of the console window when I click on “Add 
Block”. I’m also attaching a screenshot

regards,

Robert



contentscript.js:680 injection entry point. starting message loop with 
extension sandbox and waiting for settings...
js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4861 Uncaught 
ReferenceError: _ is not defined
     at r.triggerDisplace 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4861)
     at r.updateToolbarHeight 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4858)
     at p (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
     at v (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
     at u (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
     at r.a.trigger 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
     at r.set (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
     at mediaQueryChangeHandler 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4655)
DevTools failed to load SourceMap: Could not load content for 
https://dc.colliercollective.org/~ec2-user/web/libraries/civicrm/core/bower_components/angular/angular.min.js.map: 
HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
angular.min.js?r=XyWih:127 TypeError: designerDialog.$el.block is not a 
function
     at HTMLDivElement.open (eval at <anonymous> (jquery.min.js:2), 
<anonymous>:92:32)
     at t.<computed>.<computed>._trigger 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
     at t.<computed>.<computed>.open 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:1963)
     at t.<computed>.<computed>.open 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
     at t.<computed>.<computed>._init 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:1963)
     at t.<computed>.<computed>._init 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
     at t.<computed>.<computed>._createWidget 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
     at new t.<computed>.<computed> 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
     at HTMLDivElement.<anonymous> 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
     at Function.each 
(js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:2) "Possibly 
unhandled rejection: {}"
(anonymous) @ angular.min.js?r=XyWih:127
(anonymous) @ angular.min.js?r=XyWih:99
g @ angular.min.js?r=XyWih:139
$digest @ angular.min.js?r=XyWih:151
$apply @ angular.min.js?r=XyWih:155
(anonymous) @ angular.min.js?r=XyWih:198
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3


--
Robert Guerra
Cel/Tel +1 416 893 0377
Twitter: twitter.com/netfreedom
Email: ***@***.***

**Comment by colemanw** [comment]
That's difficult to read. I don't think the formatting of your email came through, nor did the attachment. Try posting directly on github. Thanks.

**Comment by privaterra** [comment]
Let me try again - am copy & pasting the error message that appears...

`contentscript.js:680 injection entry point. starting message loop with extension sandbox and waiting for settings...
js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4861 Uncaught ReferenceError: _ is not defined
    at r.triggerDisplace (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4861)
    at r.updateToolbarHeight (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4858)
    at p (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
    at v (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
    at u (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
    at r.a.trigger (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
    at r.set (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
    at mediaQueryChangeHandler (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4655)
DevTools failed to load SourceMap: Could not load content for https://dc.colliercollective.org/~ec2-user/web/libraries/civicrm/core/bower_components/angular/angular.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
angular.min.js?r=XyWih:127 TypeError: designerDialog.$el.block is not a function
    at HTMLDivElement.open (eval at <anonymous> (jquery.min.js:2), <anonymous>:92:32)
    at t.<computed>.<computed>._trigger (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at t.<computed>.<computed>.open (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:1963)
    at t.<computed>.<computed>.open (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at t.<computed>.<computed>._init (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:1963)
    at t.<computed>.<computed>._init (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at t.<computed>.<computed>._createWidget (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at new t.<computed>.<computed> (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at HTMLDivElement.<anonymous> (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at Function.each (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:2) "Possibly unhandled rejection: {}"
(anonymous) @ angular.min.js?r=XyWih:127
(anonymous) @ angular.min.js?r=XyWih:99
g @ angular.min.js?r=XyWih:139
$digest @ angular.min.js?r=XyWih:151
$apply @ angular.min.js?r=XyWih:155
(anonymous) @ angular.min.js?r=XyWih:198
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3`

**Comment by colemanw** [comment]
I wish I could reproduce this. It works fine for me on stock (civibuild) demo sites running Drupal7 and Wordpress.
Perhaps one of your installed extensions is to blame? You could try temporarily disabling other extensions and flushing caches.

**Comment by privaterra** [comment]
I have the extension working fine on a drupal 7 install. However, the  issue occurs  on the drupal 9/civicrm (installed via composer)

Might it be an API issue ?

I’ve tried flushing the cache. Doesn’t help

In the meantime, later today follow up on your suggestion to disable other extensions and see if the problem persists. I’ll report back what I find

If having access to the server would help, let me know privately.

Robert

-- Robert Guerra Cel/Tel +1 416 893 0377 Twitter: twitter.com/netfreedom Email: ***@***.***

**Comment by privaterra** [comment]
I've tried flushing the caches and leaving all but angular profiles and the contact layout extension active. I still encounter an issue when creating new blocks of any kind..

Here's, I hope a slightly better, formatted copy/paste from the Javascript console...


contentscript.js:680 injection entry point. starting message loop with extension sandbox and waiting for settings...
angular.js?r=5R5lM:15635 

**TypeError: designerDialog.$el.block is not a function**
    at HTMLDivElement.open (eval at <anonymous> (jquery.min.js:2), <anonymous>:92:32)
    at t.<computed>.<computed>._trigger (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at t.<computed>.<computed>.open (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:1963)
    at t.<computed>.<computed>.open (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at t.<computed>.<computed>._init (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:1963)
    at t.<computed>.<computed>._init (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at t.<computed>.<computed>._createWidget (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at new t.<computed>.<computed> (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at HTMLDivElement.<anonymous> (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at Function.each (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:2) "Possibly unhandled rejection: {}"
(anonymous) @ angular.js?r=5R5lM:15635
js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4861 

**Uncaught ReferenceError: _ is not defined**
    at r.triggerDisplace (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4861)
    at r.updateToolbarHeight (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4858)
    at p (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
    at v (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
    at u (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
    at r.a.trigger (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
    at r.set (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:96)
    at mediaQueryChangeHandler (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:4655)
angular.js?r=5R5lM:15635 

**TypeError: designerDialog.$el.block is not a function**
    at HTMLDivElement.open (eval at <anonymous> (jquery.min.js:2), <anonymous>:92:32)
    at t.<computed>.<computed>._trigger (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at t.<computed>.<computed>.open (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:1963)
    at t.<computed>.<computed>.open (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at t.<computed>.<computed>._init (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:1963)
    at t.<computed>.<computed>._init (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at t.<computed>.<computed>._createWidget (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at new t.<computed>.<computed> (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at HTMLDivElement.<anonymous> (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:464)
    at Function.each (js_X6lEOJqfW1sP2ERPxUEAhHLw1cpINRIvp3f59aFLwoM.js:2) "Possibly unhandled rejection: {}"
(anonymous) @ angular.js?r=5R5lM:15635
(anonymous) @ angular.js?r=5R5lM:11914
processChecks @ angular.js?r=5R5lM:18043
$digest @ angular.js?r=5R5lM:19180
$apply @ angular.js?r=5R5lM:19568
(anonymous) @ angular.js?r=5R5lM:29023
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3

**Comment by colemanw** [comment]
The formatting is fine but it doesn't show me the source of the error.
If you want to email me with a login I can take a look on your site to try to find the problem.
coleman@civicrm.org

**Comment by colemanw** [comment]
@privaterra I'm not sure, but this patch might fix your problem. Can you try it?
```diff
diff --git a/CRM/Contactlayout/Page/Angular.php b/CRM/Contactlayout/Page/Angular.php
index 3a8cc60..b6559e4 100644
--- a/CRM/Contactlayout/Page/Angular.php
+++ b/CRM/Contactlayout/Page/Angular.php
@@ -19,6 +19,9 @@ class CRM_Contactlayout_Page_Angular extends CRM_Core_Page {
       'relationshipTypes' => (array) civicrm_api4('RelationshipType', 'get', ['where' => [['is_active', '=', TRUE]]]),
     ]);
 
+    // For editing profile blocks
+    CRM_UF_Page_ProfileEditor::registerProfileScripts();
+
     // Bootstrap Angular and set page name.
     $loader = new Civi\Angular\AngularLoader();
     $loader->setPageName('civicrm/admin/contactlayout');
```

**Comment by colemanw** [comment]
I've pushed in that patch as it fixed the main issue. Your other problem seems to be a mismatch between profile contact type and custom field contact type.
